### PR TITLE
+mecab

### LINKED ISSUE #540: +mecab (160/548)
Formulae: https://formulae.brew.sh/formula/mecab

### PR DESCRIPTION
**Pull request by sanchitram1**
closes #540

### CODE REVIEW

File: projects/mecab/package.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+distributable:
+  url: https://deb.debian.org/debian/pool/main/m/mecab/mecab_{{version.raw}}.orig.tar.gz
+  # Canonical URL = https://drive.google.com/uc?export=download&id=0B4y35FiV1wh7cENtOXlicTFaRUE
+  # homepage = https://taku910.github.io/mecab/
+  strip-components: 1
+
+versions:
+  - 0.996
+  # Do not think they have versions on the site, homebrew picks it up from deb.debian.org
```
Comment (mxcl) [inline projects/mecab/package.yml]: @jhheider anywhere you can see we can scrape the versions?

Comment (sanchitram1) [reply projects/mecab/package.yml]: in the [translated web page](https://taku910.github.io/mecab/), there is a What's New section that has some versions.

**Comment by sanchitram1** [comment]
dependency for #571

File: projects/mecab/package.yml (hunk 1 above)
Comment (jhheider) [reply projects/mecab/package.yml]: http://ftp.debian.org/debian/pool/main/m/mecab/

Comment (jhheider) [reply projects/mecab/package.yml]: Package should likely be named taku910.github.io/mecab

Comment (mxcl) [reply projects/mecab/package.yml]: good catch, yeah this name is no good. We need an audit that detects such things imo.

**Comment by mxcl** [comment]
Needs a couple changes thanks @sanchitram1

File: projects/mecab/package.yml (hunk 1 above)
Comment (jhheider) [reply projects/mecab/package.yml]: shouldn't be hard. I envision a tri-partite test
- package name is parsable as a valid URL
- URL.host contains a period
- URL.host is resolvable via DNS

**Comment by sanchitram1** [comment]
> http://ftp.debian.org/debian/pool/main/m/mecab/

@jhheider Apologizing in advance, I am terrible with regex & url scraping.  On your linked webpage, the tarball I want to fetch is [mecab_0.996.orig.tar.gz](http://ftp.debian.org/debian/pool/main/m/mecab/mecab_0.996.orig.tar.gz).  So, the regex to match in this case would be: 

```yaml
url: http://ftp.debian.org/debian/pool/main/m/mecab/
match: /mecab_\d+\.\d+\.orig\.tar\.gz
```

I am getting an error.  My theory is that tea expects major, minor, and patch, but the package only provides major & minor?

```sh
error: Uncaught Error: not-found: version: taku910.github.io/mecab
  if (!version) throw new Error(`not-found: version: ${utils.pkg.str(spec)}`)
                      ^
    at Object.resolve (file:///Users/sanch/.tea/tea.xyz/brewkit/v0.34.1/lib/usePantry.ts:41:23)
    at eventLoopTick (ext:core/01_core.js:182:11)
    at async file:///Users/sanch/.tea/tea.xyz/brewkit/v0.34.1/libexec/resolve.ts:15:15
````

**Comment by jhheider** [comment]
No, that part is fine. You need an array of `strip` regexes to remove everything that isn't version info. Probably

```yaml
strip:
  - /mecab_/
  - /\.orig\.tar\.gz/
```

**Comment by sanchitram1** [comment]
I'd add the `relocatibility` commit to the examples @mfts.  I feel like I have seen it before somewhere.

**Comment by jhheider** [comment]
Yeah, it's a common fixup for shell scripts.

**Pull request merged**